backend/src/services/borrowRequestService.ts

```diff
@@ -1,127 +1,151 @@
-    user: string,
-    borrowStart: Date,
-    borrowEnd: Date
+	gameId: string,
+	user: string,
+	borrowStart: Date,
+	borrowEnd: Date
 ) => {
-    if (borrowStart < new Date(new Date().toDateString())) return BorrowRequestState.InPast;
-    if (borrowEnd < borrowStart) return BorrowRequestState.Inverted;
-    const borrowRequestStatus = await controlBorrowRequestStatus(gameId, borrowStart, borrowEnd);
-    if (borrowRequestStatus == BorrowRequestState.Free) {
-        await prisma.borrowRequest.create({
-            data: {
-                gameId,
-                user,
-                borrowStart,
-                borrowEnd
-            }
-        });
-    }
-    return borrowRequestStatus;
+	if (borrowStart < new Date(new Date().toDateString()))
+		return BorrowRequestState.InPast;
+
+	if (borrowEnd < borrowStart) return BorrowRequestState.Inverted;
+
+	const borrowRequestStatus = await controlBorrowRequestStatus(
+		gameId,
+		borrowStart,
+		borrowEnd
+	);
+
+	const gammaUser = await getAccountFromCid(user);
+
+	const userId = gammaUser!.id;
+
+	if (borrowRequestStatus == BorrowRequestState.Free) {
+		await prisma.borrowRequest.create({
+			data: {
+				gameId,
+				userId,
+				borrowStart,
+				borrowEnd
+			}
+		});
+	}
+
+	return borrowRequestStatus;
 };
 
 export const respondBorrowRequest = async (
-    gameId: string,
-    borrowStart: Date,
-    borrowEnd: Date,
-    approved: boolean
+	gameId: string,
+	borrowStart: Date,
+	borrowEnd: Date,
+	approved: boolean
 ) => {
-    let borrowRequestStatus = await controlBorrowRequestStatus(gameId, borrowStart, borrowEnd);
-    if (borrowRequestStatus == BorrowRequestState.Pending) {
-        await prisma.borrowRequest.updateMany({
-            where: {
-                gameId: gameId,
-                borrowStart: borrowStart,
-                borrowEnd: borrowEnd
-            },
-            data: {
-                status: approved ? BorrowRequestStatus.ACCEPTED : BorrowRequestStatus.REJECTED
-            }
-        });
-        borrowRequestStatus = approved ? BorrowRequestState.Approved : BorrowRequestState.Rejected;
-        let borrowRequest = await prisma.borrowRequest.findFirst({
-            where: {
-                gameId: gameId,
-                borrowStart: borrowStart,
-                borrowEnd: borrowEnd
-            }
-        })
-        if (borrowRequest === null) return BorrowRequestState.NotValid;
-    }
-    return borrowRequestStatus;
-}
+	let borrowRequestStatus = await controlBorrowRequestStatus(
+		gameId,
+		borrowStart,
+		borrowEnd
+	);
+	if (borrowRequestStatus == BorrowRequestState.Pending) {
+		await prisma.borrowRequest.updateMany({
+			where: {
+				gameId: gameId,
+				borrowStart: borrowStart,
+				borrowEnd: borrowEnd
+			},
+			data: {
+				status: approved
+					? BorrowRequestStatus.ACCEPTED
+					: BorrowRequestStatus.REJECTED
+			}
+		});
+		borrowRequestStatus = approved
+			? BorrowRequestState.Approved
+			: BorrowRequestState.Rejected;
```

Comment:
This should just be done on the return if you do the change my previous comment says